sdk/samples/core/mosquitto_rpc_server_sample.c

```diff
@@ -97,7 +98,7 @@ static void timer_callback(union sigval sv)
           .status = AZ_MQTT5_RPC_STATUS_TIMEOUT,
           .response = AZ_SPAN_EMPTY,
           .content_type = AZ_SPAN_EMPTY };
-  if (az_result_failed(az_mqtt5_rpc_server_execution_finish(&rpc_server, &return_data)))
+  if (az_result_failed(az_mqtt5_rpc_server_execution_finish(&rpc_server_policy, &return_data)))
   {
     printf(LOG_APP_ERROR "Failed sending execution response to HFSM\n");
     return;
```

Comment:
Another point is that this sample lacks thread safety. I do not believe the hsfm is thread safe, so we should make sure that that is addressed here. Also it would simplify these data copies and later checks.

Reply:
Will address this in a future PR since the issue isn't introduced in this one